File: back_end/RB120-129/lesson_4/easy_3/1.rb
# If we have this code:

class Greeting
  def greet(message)
    puts message
  end
end

class Hello < Greeting
  def hi
    greet("Hello")
  end
end

class Goodbye < Greeting
  def bye
    greet("Goodbye")
  end
end

# What happens in each of the following cases:

hello = Hello.new
puts hello.hi # Hello

hello = Hello.new
puts hello.bye # error, no method 'bye' for hello class

hello = Hello.new
puts hello.greet # error, no arguments passed to greet

hello = Hello.new
puts hello.greet("Goodbye") # Goodbye

puts Hello.hi # error, no class method defined

File: exercises/javascript_basics/functions_2/1.js
// function greet(greeting) {
//   console.log(greeting + ', world!');
// }
//
// console.log(greet('Salutations'));
// console.log(greet());

function greet(greeting = 'Hello') {
  console.log(greeting + ', world!');
}

console.log(greet('Salutations'));
console.log(greet(undefined));
